redfish-core/lib/license_service.hpp

```diff
@@ -99,38 +99,34 @@ inline void
 
     if (status == "com.ibm.License.LicenseManager.Status.ActivationFailed")
     {
-        // TODO: Need to return appropriate redfish error
         BMCWEB_LOG_ERROR << "LicenseActivationStatus: ActivationFailed";
-        messages::internalError(asyncResp->res);
+        messages::installFailed(asyncResp->res);
     }
     else if (status == "com.ibm.License.LicenseManager.Status.InvalidLicense")
     {
-        // TODO: Need to return appropriate redfish error
         BMCWEB_LOG_ERROR << "LicenseActivationStatus: InvalidLicense";
-        messages::internalError(asyncResp->res);
+        messages::invalidLicense(asyncResp->res);
     }
     else if (status == "com.ibm.License.LicenseManager.Status.IncorrectSystem")
     {
-        // TODO: Need to return appropriate redfish error
         BMCWEB_LOG_ERROR << "LicenseActivationStatus: IncorrectSystem";
-        messages::internalError(asyncResp->res);
+        messages::notApplicableToTarget(asyncResp->res);
     }
     else if (status ==
              "com.ibm.License.LicenseManager.Status.IncorrectSequence")
     {
-        // TODO: Need to return appropriate redfish error
         BMCWEB_LOG_ERROR << "LicenseActivationStatus: IncorrectSequence";
-        messages::internalError(asyncResp->res);
+        messages::notApplicableToTarget(asyncResp->res);
     }
     else if (status == "com.ibm.License.LicenseManager.Status.InvalidHostState")
     {
-        // TODO: Need to return appropriate redfish error
         BMCWEB_LOG_ERROR << "LicenseActivationStatus: InvalidHostState";
         messages::internalError(asyncResp->res);
```

Comment:
This can be an InstallFailed error , with "InvalidHostState" as the argument

Reply:
done